addons/website_sale/static/src/js/extra_info.js

```diff
@@ -1,76 +1,45 @@
-import publicWidget from "@web/legacy/js/public/public_widget";
+import { Interaction } from "@web/public/interaction";
+import { registry } from "@web/core/registry";
 
-publicWidget.registry.websiteSaleExtraInfo = publicWidget.Widget.extend({
-    // /shop/extra_info
-    selector: '#class_extra_info',
 
-    /**
-     * @constructor
-     */
-    init: function () {
-        this._super.apply(this, arguments);
-
-        this.http = this.bindService('http');
-
-        this.submitButton =  document.getElementsByName('website_sale_main_button')[0]
-        this._boundSaveExtraInfo = this._onSaveExtraInfo.bind(this);
-        this.submitButton.addEventListener("click", this._boundSaveExtraInfo);
-        this.extraForm = document.querySelector('form.form_extra_info');
-    },
-
-    destroy() {
-        this.submitButton.removeEventListener("click", this._boundSaveExtraInfo);
-        this._super(...arguments);
-    },
-
-    //--------------------------------------------------------------------------
-    // Handlers
-    //--------------------------------------------------------------------------
+export class WebsiteSaleExtraInfo extends Interaction {
+    static selector = "#class_extra_info";
+    dynamicSelectors = {
+        _submitbutton: () => document.querySelector('[name="website_sale_main_button"]'),
+    };
+    dynamicContent = {
+        _submitbutton: {"t-on-click.withTarget": this.onSaveExtraInfo},
```

Comment:
I'm not quite sure the `withTarget` is needed here. Can't we use `ev.currentTarget` in `onSaveExtraInfo`? See https://gist.github.com/ged-odoo/62810b496cc95e798fa0f0c613013fb6#event-handling
```suggestion
        _submitbutton: {"t-on-click": this.onSaveExtraInfo},
```

Reply:
yes, you are right. `withTarget` is basically only useful for some rare cases with async handlers